src/node.js

```diff
@@ -0,0 +1,70 @@
+module.exports = HeimdallNode;
+function HeimdallNode(heimdall, id, data, parent) {
+  this.heimdall = heimdall;
+
+  this.id = id;
+  this._id = heimdall._nextId++;
+  this.stats = this.heimdall._createStats(data);
+  this.children = [];
+  this.parent = parent;
+}
+
+Object.defineProperty(HeimdallNode.prototype, 'isRoot', {
+  get: function () {
+    return this.parent === undefined;
+  },
+});
+
+HeimdallNode.prototype.visitPreOrder = function (cb) {
+  cb(this);
+
+  for (var i = 0; i < this.children.length; i++) {
+    this.children[i].visitPreOrder(cb);
+  }
+};
+
+HeimdallNode.prototype.visitPostOrder = function (cb) {
+  for (var i = 0; i < this.children.length; i++) {
+    this.children[i].visitPostOrder(cb);
+  }
+
+  cb(this);
+};
+
+HeimdallNode.prototype.remove = function () {
+  if (!this.parent) {
+    throw new Error('Cannot remove the root heimdalljs node.');
+  }
+  if (this.heimdall.current === this) {
+    throw new Error('Cannot remove an active heimdalljs node.');
+  }
+
+  var index = this.parent.children.indexOf(this);
```

Comment:
We may want to throw if `index < 0`, although defensive it think it will save future trolling